Loop/Managers/DeviceDataManager.swift

```diff
@@ -318,20 +389,45 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
 
         device.assertIdleListening()
 
-        // How long should we wait before we poll for new reservoir data?
-        let reservoirTolerance = rileyLinkManager.idleListeningEnabled ? NSTimeInterval(minutes: 11) : NSTimeInterval(minutes: 4)
+        // How long should we wait before we poll for new pump data?
+        let pumpStatusAgeTolerance = rileyLinkManager.idleListeningEnabled ? NSTimeInterval(minutes: 11) : NSTimeInterval(minutes: 4)
 
-        // If we don't yet have reservoir data, or it's old, poll for it.
-        if latestReservoirValue == nil || latestReservoirValue!.startDate.timeIntervalSinceNow <= -reservoirTolerance {
-            readReservoirVolume { (result) in
+        // If we don't yet have pump status, or it's old, poll for it.
+        if latestPumpStatusDate == nil || latestPumpStatusDate!.timeIntervalSinceNow <= -pumpStatusAgeTolerance {
+            guard let device = rileyLinkManager.firstConnectedDevice else {
+                return
+            }
+
+            guard let ops = device.ops else {
+                self.troubleshootPumpCommsWithDevice(device)
+                return
+            }
+
+            ops.readPumpStatus({ (result) in
                 switch result {
-                case .Success(let (units, date)):
-                    self.updateReservoirVolume(units, atDate: date, withTimeLeft: nil)
+                case .Success(let status):
+                    self.latestPolledPumpStatus = status
+
+                    let battery = BatteryStatus(voltage: status.batteryVolts, status: String(status.batteryStatus).lowercaseString)
```

Comment:
> String(status.batteryStatus).lowercaseString

Is this intended to be i18n friendly? or is it a Nightscout enumeration of some kind? If the former, then it should be adjusted to pull from the strings bundle. If the latter, then best to keep it internal to `NightscoutUploadKit` and surface a native `enum`.